### PR TITLE
Add async generator logic to ProcessPromise

### LINKED ISSUE #965: feat: make `ProcessPromise` iterable via async for
Imagine:
```ts
const p = $`cmd`
for await (const chunk of p) {
  // ...
}

const o = await $`cmd`
for (const chunk of o) {
  // ...
}
```
Keep in mind, It is important to make iterator available at any stage of the process. See how the `pipe()` API handles this requirement.

### PR DESCRIPTION
**Pull request by mxmvshnvsk**
Fixes #965

### CODE REVIEW

File: test/core.test.js [hunk 1]
```diff
@@ -703,6 +703,36 @@ describe('core', () => {
...
+          lines.push(line)
+        }
+    
+        assert.equal(lines.length, 3, 'Should have 3 lines')
+        assert.equal(lines[0], 'PartialLine1', 'First line should be "PartialLine1"')
+        assert.equal(lines[1], 'Line2', 'Second line should be "Line2"')
+        assert.equal(lines[2], 'PartialLine3', 'Third line should be "PartialLine3"')
+      })
+    })
+
```
Comment (antonmedv) [inline test/core.test.js:735]: Lets add test for empty and single line (without newline char at the end) stdouts.

Comment (mxmvshnvsk) [reply test/core.test.js:758]: Done

File: src/core.ts [hunk 2]
```diff
@@ -543,6 +543,31 @@ export class ProcessPromise extends Promise<ProcessOutput> {
     return super.catch(onrejected)
   }
 
+  // Async iterator API
+  async *[Symbol.asyncIterator]() {
```
Comment (antongolub) [inline src/core.ts:547]: Imagine:
```ts
const p = $`super-long-cmd`

setTimeout(() => {
   for await (const chunk of p) {
      console.log(chunk) // all previously stored and all next chunks should be available here. See internal `fill()` helper of `ProcessPromise.pipe()`
   }
}, 1000)
```

**Review by antongolub — CHANGES_REQUESTED** [review]

**Review by antongolub — APPROVED** [review]
Seems good! Thanks for your efforts.
